core/src/main/java/nl/nn/adapterframework/core/ProcessState.java

```diff
@@ -50,4 +62,35 @@
 		}
 		return targetProcessStates;
 	}
+
+	public static ProcessState getProcessStateFromName(String name) {
+		ProcessState[] processStates = ProcessState.values();
+		for (ProcessState processState : processStates) {
+			if(StringUtils.equalsIgnoreCase(processState.getName(), name)) {
+				return processState;
+			}
+		}
+		return null;
+	}
+
+	public void setIconName(String iconName) {
+		this.iconName = iconName;
+	}
+	public String getIconName() {
+		return iconName;
+	}
+
+	public void setType(String type) {
+		this.type = type;
+	}
+	public String getType() {
+		return type;
+	}
+
+	public void setName(String name) {
+		this.name = name;
+	}
+	public String getName() {
+		return name;
+	}
```

Comment:
You could replace these getters and setters with Lombok field annotations ``@Getter`` and ``@Setter``